blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -559,4 +612,91 @@ public Boolean apply(Row<Locator, Long> row) {
         return excessEnumMetrics;
     }
 
+    private <T extends Rollup> Points<T> getEnumStringValuesFromHashes(Points<T> points, ColumnList<Long> enumvalues) {
+        Map<Long, String> hash2enumValues = getEnumValueFromHash(enumvalues);
+        Points<T> pointsEnum = new Points<T>();
+
+        Map<Long, Points.Point<T>> pointsMap = points.getPoints();
+
+        for (Long timestamp : points.getPoints().keySet()) {
+            BluefloodEnumRollup enumRollup = (BluefloodEnumRollup)pointsMap.get(timestamp).getData();
+            for (Long hash : enumRollup.getHashedEnumValuesWithCounts().keySet()) {
+                String enumValueString = hash2enumValues.get(hash);
+                enumRollup.getStringEnumValuesWithCounts().put(enumValueString, enumRollup.getHashedEnumValuesWithCounts().get(hash));
+                pointsEnum.add(new Points.Point<T>(timestamp,(T)enumRollup));
+            }
+        }
+        return pointsEnum;
+    }
+
+    private Map<Long, String> getEnumValueFromHash(ColumnList<Long> enumValues) {
+        HashMap<Long,String> hash2enumValues = new HashMap<Long, String>();
+
+        for (Column<Long> col: enumValues) {
+            hash2enumValues.put(col.getName(), col.getStringValue());
+        }
+
+        return hash2enumValues;
+    }
+
+    private ColumnList<Long> getColumnsFromEnumCF(final Locator locator) {
+        final Map<Locator, ColumnList<Long>> columns = new HashMap<Locator, ColumnList<Long>>();
+
+        try {
+            OperationResult<Rows<Locator, Long>> query = getKeyspace()
+                    .prepareQuery(CassandraModel.CF_METRICS_ENUM)
+                    .getKeySlice(locator)
+                    .execute();
+
+            for (Row<Locator, Long> row : query.getResult()) {
+                columns.put(row.getKey(), row.getColumns());
+            }
+        } catch (ConnectionException e) {
+            if (e instanceof NotFoundException) { // TODO: Not really sure what happens when one of the keys is not found.
+                System.out.println("CF Not found");
+            } else {
+                System.out.println("Some other error");
+            }
+        } finally {
+        }
+        return columns.get(locator);
+    }
+
+    class ColumnFromEnumCF implements Callable{
```

Comment:
I am not a fan of inner classes.  It makes the top AstyanaxReader class very bloated, and hard to reuse in the future (as you can see from all the class extraction refactoring we have had to do recently).  Any reason why this and the PointsFromDB class below can't be standalone classes in the com.rackspacecloud.blueflood.io package?